File: scripts/fix-imports.py
#!/usr/bin/env python3
import re
import sys
from pathlib import Path

def fix_imports_in_file(filepath):
    """Fix missing imports in a TypeScript/TSX file"""
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    original_content = content
    modified = False
    
    # Check if file uses React hooks but doesn't import them
    react_hooks = ['useState', 'useEffect', 'useMemo', 'useCallback', 'useRef']
    lucide_icons = ['AlertCircle', 'Loader2']
    
    # Find existing React import
    react_import_match = re.search(r"import\s+{([^}]+)}\s+from\s+['\"]react['\"]", content)
    
    if react_import_match:
        existing_imports = set(i.strip() for i in react_import_match.group(1).split(','))
        needed_imports = set()
        
        for hook in react_hooks:
            # Check if hook is used but not imported
            if re.search(rf'\b{hook}\s*\(', content) and hook not in existing_imports:
                needed_imports.add(hook)
        
        if needed_imports:
            all_imports = sorted(existing_imports | needed_imports)
            new_import = f"import {{ {', '.join(all_imports)} }} from 'react'"
            content = content.replace(react_import_match.group(0), new_import)
            modified = True
    
    # Find existing lucide-react import
    lucide_import_match = re.search(r"import\s+{([^}]+)}\s+from\s+['\"]lucide-react['\"]", content)
    
    if lucide_import_match:
        existing_imports = set(i.strip() for i in lucide_import_match.group(1).split(','))
        needed_imports = set()
        
        for icon in lucide_icons:
            # Check if icon is used but not imported
            if f'<{icon}' in content and icon not in existing_imports:
                needed_imports.add(icon)
        
        if needed_imports:
            all_imports = sorted(existing_imports | needed_imports)
            new_import = f"import {{ {', '.join(all_imports)} }} from 'lucide-react'"
            content = content.replace(lucide_import_match.group(0), new_import)
            modified = True
    
    if modified and content != original_content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        return True
    return False

if __name__ == '__main__':
    files_to_fix = [
        'src/app/compvss/issues/routing/page.tsx',
        'src/app/compvss/operations/check-in/page.tsx',
        'src/app/compvss/operations/tasks/page.tsx',
        'src/app/compvss/settings/account/page.tsx',
        'src/app/compvss/team/directory/page.tsx',
        'src/app/compvss/team/members/page.tsx',
        'src/app/gvteway/adventures/page.tsx',
        'src/app/gvteway/auth/verify-email/page.tsx',
        'src/app/gvteway/events/calendar/page.tsx',
        'src/app/gvteway/events/category/[slug]/page.tsx',
        'src/app/gvteway/marketplace/page.tsx',
        'src/app/gvteway/settings/account/page.tsx',
        'src/app/gvteway/settings/page.tsx',
        'src/app/gvteway/social/page.tsx',
        'src/app/gvteway/tickets/page.tsx',
        'src/app/gvteway/wallet/nft/page.tsx',
        'src/app/gvteway/wallet/page.tsx',
        'src/components/atlvs/DataTable.tsx',
        'src/hooks/useABTest.tsx',
    ]
    
    base_path = Path('/Users/julianclarkson/Documents/Grasshopper26.10')
    fixed_count = 0
    
    for file_path in files_to_fix:
        full_path = base_path / file_path
        if full_path.exists():
            if fix_imports_in_file(full_path):
                print(f"Fixed: {file_path}")
                fixed_count += 1
        else:
            print(f"Not found: {file_path}")
    
    print(f"\nFixed {fixed_count} files")

File: scripts/fix-imports.mjs
#!/usr/bin/env node

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const rootDir = path.resolve(__dirname, '..');

// Map of files that need z import
const filesNeedingZod = [
  'src/app/api/account/delete/route.ts',
  'src/app/api/atlvs/advancing/[id]/comments/route.ts',
  'src/app/api/atlvs/advancing/[id]/route.ts',
  'src/app/api/atlvs/advancing/[id]/status/route.ts',
  'src/app/api/atlvs/inventory/route.ts',
  'src/app/api/atlvs/kpi/event/[eventId]/route.ts',
  'src/app/api/atlvs/kpi/marketing/[eventId]/route.ts',
  'src/app/api/atlvs/maintenance/route.ts',
  'src/app/api/atlvs/opportunities/[id]/applications/route.ts',
  'src/app/api/atlvs/opportunities/[id]/route.ts',
  'src/app/api/atlvs/opportunities/route.ts',
  'src/app/api/atlvs/projects/route.ts',
  'src/app/api/atlvs/settings/route.ts',
  'src/app/api/atlvs/workflows/route.ts',
  'src/app/api/auth/refresh/route.ts',
  'src/app/api/batch/emails/route.ts',
  'src/app/api/batch/notifications/route.ts',
  'src/app/api/batch/qr-codes/route.ts',
  'src/app/api/compvss/affiliates/route.ts',
  'src/app/api/compvss/applications/route.ts',
  'src/app/api/compvss/assets/checkout/route.ts',
  'src/app/api/compvss/assets/route.ts',
  'src/app/api/compvss/documents/route.ts',
  'src/app/api/events/stream/route.ts',
  'src/app/api/integrations/push/route.ts',
  'src/app/api/integrations/slack/route.ts',
  'src/app/api/profile/route.ts',
  'src/app/api/social/follow/route.ts',
  'src/app/api/spotify/artists/[id]/route.ts',
  'src/app/api/tickets/purchase/route.ts',
  'src/app/api/wishlists/route.ts',
];

// Map of files that need prisma import
const filesNeedingPrisma = [
  'src/app/api/account/delete/route.ts',
  'src/app/api/adventures/[id]/book/route.ts',
  'src/app/api/adventures/route.ts',
  'src/app/api/alerts/route.ts',
  'src/app/api/artists/[id]/route.ts',
  'src/app/api/artists/route.ts',
  'src/app/api/atlvs/advancing/[id]/route.ts',
  'src/app/api/atlvs/analytics/insights/route.ts',
  'src/app/api/atlvs/automation/route.ts',
  'src/app/api/atlvs/budgets/route.ts',
  'src/app/api/atlvs/equipment/route.ts',
  'src/app/api/atlvs/inventory/route.ts',
  'src/app/api/atlvs/maintenance/route.ts',
  'src/app/api/atlvs/projects/route.ts',
  'src/app/api/atlvs/tasks/route.ts',
  'src/app/api/atlvs/teams/route.ts',
  'src/app/api/atlvs/workflows/route.ts',
  'src/app/api/auth/[...nextauth]/route.ts',
  'src/app/api/auth/forgot-password/route.ts',
  'src/app/api/auth/login/route.ts',
  'src/app/api/auth/logout/route.ts',
  'src/app/api/auth/me/route.ts',
  'src/app/api/auth/refresh-token/route.ts',
  'src/app/api/auth/refresh/route.ts',
  'src/app/api/auth/register/route.ts',
  'src/app/api/auth/resend-verification/route.ts',
  'src/app/api/auth/reset-password/route.ts',
  'src/app/api/auth/session/route.ts',
  'src/app/api/auth/verify-email/route.ts',
  'src/app/api/auth/wallet/route.ts',
  'src/app/api/batch/qr-codes/route.ts',
  'src/app/api/cart/items/route.ts',
  'src/app/api/compvss/advancing/route.ts',
  'src/app/api/compvss/affiliates/route.ts',
  'src/app/api/compvss/assets/checkout/route.ts',
  'src/app/api/compvss/assets/route.ts',
  'src/app/api/compvss/documents/route.ts',
  'src/app/api/compvss/expenses/route.ts',
  'src/app/api/compvss/issues/route.ts',
  'src/app/api/compvss/tasks/route.ts',
  'src/app/api/crypto/payment/route.ts',
  'src/app/api/events/featured/route.ts',
  'src/app/api/events/route.ts',
  'src/app/api/memberships/me/route.ts',
  'src/app/api/memberships/tiers/route.ts',
  'src/app/api/nft/mint/route.ts',
  'src/app/api/notifications/route.ts',
  'src/app/api/orders/route.ts',
  'src/app/api/organizations/route.ts',
  'src/app/api/products/route.ts',
  'src/app/api/profile/settings/route.ts',
  'src/app/api/social/follow/route.ts',
  'src/app/api/social/posts/[id]/comments/route.ts',
  'src/app/api/social/posts/[id]/like/route.ts',
  'src/app/api/social/posts/route.ts',
  'src/app/api/sync/compvss-to-atlvs/route.ts',
  'src/app/api/sync/gvteway-to-atlvs/route.ts',
  'src/app/api/tickets/[id]/route.ts',
  'src/app/api/tickets/[id]/validate/route.ts',
  'src/app/api/tickets/purchase/route.ts',
  'src/app/api/tickets/route.ts',
  'src/app/api/tickets/validate/route.ts',
  'src/app/api/venues/[id]/route.ts',
  'src/app/api/venues/route.ts',
  'src/app/api/webhooks/stripe/route.ts',
];

// Files needing getClientIdentifier
const filesNeedingGetClientIdentifier = [
  'src/app/api/adventures/route.ts',
  'src/app/api/artists/route.ts',
  'src/app/api/auth/forgot-password/route.ts',
  'src/app/api/auth/login/route.ts',
  'src/app/api/auth/register/route.ts',
  'src/app/api/auth/reset-password/route.ts',
  'src/app/api/events/featured/route.ts',
  'src/app/api/events/route.ts',
  'src/app/api/social/posts/[id]/route.ts',
  'src/app/api/webhooks/n8n/events/route.ts',
];

// Files needing rate limit imports
const filesNeedingRateLimit = [
  'src/app/api/atlvs/kpi/event/[eventId]/route.ts',
  'src/app/api/atlvs/kpi/marketing/[eventId]/route.ts',
  'src/app/api/webhooks/sendgrid/route.ts',
  'src/app/api/webhooks/twilio/route.ts',
];

function addImportIfMissing(content, importStatement) {
  if (content.includes(importStatement)) {
    return content;
  }
  
  // Find the last import statement
  const lines = content.split('\n');
  let lastImportIndex = -1;
  
  for (let i = 0; i < lines.length; i++) {
    if (lines[i].trim().startsWith('import ') || lines[i].trim().startsWith('import{')) {
      lastImportIndex = i;
    } else if (lastImportIndex !== -1 && lines[i].trim() === '') {
      break;
    }
  }
  
  if (lastImportIndex === -1) {
    // No imports found, add at the beginning
    return importStatement + '\n' + content;
  }
  
  // Insert after the last import
  lines.splice(lastImportIndex + 1, 0, importStatement);
  return lines.join('\n');
}

function fixFile(filePath, fixes) {
  const fullPath = path.join(rootDir, filePath);
  
  if (!fs.existsSync(fullPath)) {
    console.log(`⚠️  File not found: ${filePath}`);
    return false;
  }
  
  let content = fs.readFileSync(fullPath, 'utf-8');
  let modified = false;
  
  for (const importStatement of fixes) {
    const newContent = addImportIfMissing(content, importStatement);
    if (newContent !== content) {
      content = newContent;
      modified = true;
    }
  }
  
  if (modified) {
    fs.writeFileSync(fullPath, content, 'utf-8');
    console.log(`✅ Fixed: ${filePath}`);
    return true;
  }
  
  return false;
}

let fixedCount = 0;

// Fix zod imports
console.log('\n📦 Adding zod imports...');
for (const file of filesNeedingZod) {
  if (fixFile(file, ["import { z } from 'zod';"])) {
    fixedCount++;
  }
}

// Fix prisma imports
console.log('\n📦 Adding prisma imports...');
for (const file of filesNeedingPrisma) {
  if (fixFile(file, ["import { prisma } from '@/lib/prisma';"])) {
    fixedCount++;
  }
}

// Fix getClientIdentifier imports
console.log('\n📦 Adding getClientIdentifier imports...');
for (const file of filesNeedingGetClientIdentifier) {
  if (fixFile(file, ["import { getClientIdentifier } from '@/lib/api/middleware';"])) {
    fixedCount++;
  }
}

// Fix rate limit imports
console.log('\n📦 Adding rate limit imports...');
for (const file of filesNeedingRateLimit) {
  if (fixFile(file, [
    "import { rateLimit, requireAuth } from '@/lib/api/middleware';",
    "import { RateLimitIdentifiers, RATE_LIMITS } from '@/lib/api/rate-limits';"
  ])) {
    fixedCount++;
  }
}

console.log(`\n✨ Fixed ${fixedCount} files`);
